### DISCUSSION TITLE
Local development setup

### DISCUSSION THREAD

**Discussion by starsep** [joffrey-bion/seven-wonders#134] [Q&A] [answered]
Hi! Cool project!
I'd like to contribute but I struggle to run development server with hot reload.

I tried running `./gradlew :sw-server:bootRun` and `./gradlew :sw-ui:browserDevelopmentRun` in parallel.
I think this starts Spring Boot server on port 8000 and Webpack Dev Server on port 8080.
I assume I should visit `http://localhost:8080` in the browser.

Once I make some changes e.g. card title in CardImage.kt code gets recompiled (I see that in `:sw-ui:browserDevelopmentRun` log) but no changes in the browser.
Logs from browser console:
```
[WDS] App updated. Recompiling...
[WDS] Nothing changed.
```

Please tell me what I'm doing wrong and preferably add some simple instructions for a developer in README.md.

**Comment by joffrey-bion** [comment, marked as answer]
Hi! Thanks a lot for reaching out!

You did nothing wrong, everything is exactly as you said.

The hot reloading just doesn't seem to work consistently, and I simply don't rely on it at all.
I just stop-and-rerun `./gradlew :sw-ui:browserDevelopmentRun` when I make client-only changes.
Maybe I did something wrong in the setup myself.

To be honest I did not investigate too much in this respect, because even if reloading is properly triggered, there is currently no mechanism to restore the client state (nor the session on server-side for that matter, because it is currently linked to the websocket connection).
So you could only work on the home page with hot reloading, all other pages would require you to go back to the home page and reconnect to the game (see https://github.com/joffrey-bion/seven-wonders/issues/118).

This reconnection ability is something I should work on, it is tracked here: https://github.com/joffrey-bion/seven-wonders/issues/10

I could definitely add some instructions for local development, I just didn't expect people would want to contribute so far 😄 

PS: how did you find this project, by the way?

**Reply by starsep** [reply]
> PS: how did you find this project, by the way?

I was looking for 7 Wonders bots/implementation/simulation on GitHub.
I found this project and I was happily surprised that it is implemented in my favourite programming language - Kotlin.
